app/game/starter_level.rb

```diff
@@ -38,20 +38,29 @@ def init_objects
 def tick args
   $args = args
   init(args) if args.state.tick_count == 0
+  if args.inputs.last_active == :controller
+    args.state.active_input = args.inputs.controller_one
+    args.state.input_mode = :controller
+  elsif args.inputs.last_active == :keyboard
+    args.state.active_input = args.inputs.keyboard
+    args.state.input_mode = :keyboard
+  end
```

Comment:
I had originally hoped to just use `args.inputs.last_active` here, but wasn't sure what to do when the value of that was `:mouse`. I think the safest assumption is probably that someone playing with a mouse is playing with a keyboard, but for the sake of keeping things simple I just left it at - controller and keyboard will swap each other whenever they go active, and mouse will just not modify this.

..I'd still like to eventually sort this out though because it's itching me to need an extra `args.state` value. And I'd love to not be storing `args.inputs.keyboard` or `args.inputs.controller_one` *into* `args.state.active_input`, but I needed to in order to have it be preserved if the next tick was a mouse tick.